src/System.Net.HttpListener/src/System/Net/Windows/AsyncRequestContext.cs

```diff
@@ -62,19 +91,26 @@ protected override void Dispose(bool disposing)
                 Debug.Assert(!disposing, "AsyncRequestContext::Dispose()|Must call ReleasePins() before calling Dispose().");
                 if (!Environment.HasShutdownStarted || disposing)
                 {
+#if DEBUG
+                    DebugRefCountReleaseNativeOverlapped();
+#endif
                     _boundHandle.FreeNativeOverlapped(_nativeOverlapped);
                 }
             }
+
             base.Dispose(disposing);
         }
 
         internal NativeOverlapped* NativeOverlapped
         {
             get
             {
+#if DEBUG
+                Debug.Assert(Interlocked.Increment(ref _nativeOverlappedUsed) == 1, "NativeOverlapped reused.");
+#endif
```

Comment:
FWIW, it's odd to me doing this increment in the getter. That might work for current access patterns, but it seems very brittle; calling code could reasonably expect to use the getter multiple times for the same operation.